share/picongpu/tests/Pusher/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,364 @@
+        # compare the phase after 1 periodic time
+        compare_phases = (
+            abs(theta[index_of_revolution] - theta[index_of_revolution
+                + int(steps_per_revolution/periode)]))
+        # compare with theoretical, i.e. stationary phase
+        compare_ana_phases = (
+            abs(theta[index_of_revolution] - np.pi/2 + 2*np.pi/100))
+
+        compare_result_phases = 0
+
+        if compare_phases > delta:
+            print("pusher is not valid (phases)")
+            print("please check the simulation")
+            compare_result_phases += 1
+
+        if compare_ana_phases > (k+1)*delta:
+            print("pusher is not valid (phases_analytical)")
+            print("please check the simulation")
+            compare_result_phases += 2
+
+        else:
+            print("pusher is valid (phases)")
+    return compare_result_phases
+
+
+def main(dataPath):
+    """ Main function. Calls the functions to compare PIConGPU's simulation
+    results with respect to the change of gyration radius and phase after
+    one revolution.
+
+    Returns:
+    compare_result(int): 0 if the simulated result of phase and gyration radius
+                         can be regarded as stationary
+                         Non-zero if they diverge.
+
+    """
+    # read parameters of series
+    series, params = get_params(dataPath)
+
+    B = 50
+    # extract all relevant data
+    (x_poss, y_poss, x_offSet, y_offSet, x_momentum, y_momentum, charge,
+     mass, num_iter, periode) = read_series(series)
+
+    # all radii in the series
+    radius = (np.sqrt(x_momentum**2 + y_momentum**2) * params["unit_mass"]
+              * params["unit_speed"] /
+              (abs(charge[0]) * params["unit_charge"] * B) /
+              params["cell_depth"] / params["unit_length"])
+    R_c = radius[0]  # original radius
+
+    # transformation for the correct calculation of the radii by position
+    x_poss = x_poss + (R_c * np.sin(2/100 * np.pi))
+    y_poss = y_poss - R_c + int(R_c) + (R_c - R_c * np.cos(2/100 * np.pi))
+
+    x_offSet = x_offSet - 32
+    y_offSet = y_offSet - 5 - int(R_c)
+
+    # tests
+    compare_result_radius = compare_radius(x_poss, y_poss, x_offSet, y_offSet,
+                                           x_momentum, y_momentum, charge,
+                                           mass, B, params, num_iter, periode)
+    compare_result_phases = compare_phases(x_poss, y_poss, x_offSet, y_offSet,
+                                           x_momentum, y_momentum, charge,
+                                           mass, B, params, num_iter, periode)
+
+    # yield both tests/comparisions a positive result?
+    compare_result = compare_result_radius + compare_result_phases
+
+    return compare_result
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 2:
+        raise SystemExit(f"Usage: {sys.argv[0]} <path_to_simulation_data>")
+    main(sys.argv[1])
```

Comment:
CAUTION: Currently the test can never fail!
```suggestion
    sys.exit(main(sys.argv[1]))
```

Reply:
I used 
`else:
    main(sys.argv[1])`

Reply:
No, that's not what we're fixing here: Your bash script roughly runs
```bash
python3 ./MainTest.py
exit "$?"
```
but `$?` always holds `0` (i.e. `EXIT_SUCCESS`) because no matter what the return value of `main()` is the _`python3` executable ran just fine_. You have to explicitly instruct `Python` in your script to exit with a non-zero exit code, hence
```python3
 sys.exit(main(sys.argv[1]))
```